app/forms/field_handlers/field_handler.py

```diff
@@ -18,23 +21,38 @@ class FieldHandler(ABC):
     def __init__(
         self,
         answer_schema: dict,
-        error_messages: dict = None,
+        schema: QuestionnaireSchema,
         answer_store: AnswerStore = None,
+        list_store: ListStore = None,
         metadata: dict = None,
         location: Location = None,
         disable_validation: bool = False,
         question_title: str = None,
     ):
+        self.schema = schema
         self.answer_schema = answer_schema
-        self.error_messages = error_messages or {}
+        self.error_messages = schema.error_messages or {}
         self.answer_store = answer_store or AnswerStore()
+        self.list_store = list_store or ListStore()
         self.metadata = metadata or {}
         self.location = location
         self.disable_validation = disable_validation
         self.question_title = str(question_title)
+
         self.validation_messages = self.answer_schema.get("validation", {}).get(
             "messages", {}
         )
+        list_item_id = location.list_item_id if location else None
+
+        self.value_source_resolver = ValueSourceResolver(
+            answer_store=self.answer_store,
+            list_store=self.list_store,
+            metadata=self.metadata,
+            schema=self.schema,
+            location=self.location,
+            list_item_id=list_item_id,
+            escape_answer_values=True,
+        )
```

Comment:
This is currently initialised for every answer and for every option that has a detail answer which is wasteful. We should make this a cached property then it will only be initialised once per answer and only when it is required. 

Even with the cached property we are still initialising it per answer when needed instead of per block but changing that behaviour requires further refactoring i.e not passing schema in but a value source resolver to handlers, therefore that is something to look at in future.